api/paidAction/buyCredits.js

```diff
@@ -0,0 +1,25 @@
+// XXX we don't use this yet ...
+// it's just showing that even buying credits
+// can eventually be a paid action
+
+import { USER_ID } from '@/lib/constants'
+
+export const anonable = false
+export const supportsPessimism = false
+export const supportsOptimism = true
```

Comment:
Pessimism means "use a hold invoice" and only "do stuff" when it's paid. `BUY_CREDITS` is a weird one because it's optimistic and **only** has side effects, ie `onPaid`, ie incrementing the stacker's balance, because `perform` is just making an invoice which `paidAction/index.js` does for us.